### PR TITLE
update guide for new api

### PR DESCRIPTION
**Pull request by blampe**
@breerly @abhinav @junchaowu

### CODE REVIEW

File: GUIDE.rst [hunk 1]
```diff
@@ -209,26 +211,17 @@ define our handlers as coroutines and yield to IO operations:
...
 
-    @app.register(KeyValue)
+    @tchannel.register(KeyValue)
     @gen.coroutine
-    def setValue(request, response):
-        key = request.args.key
-        value = request.args.value
+    def setValue(request):
+        key = request.body.key
+        value = request.body.value
```
Comment (abhinav) [inline GUIDE.rst]: I would make it `args = request.body` and then `args.key` and `args.value`. I want to make it clear that the method parameters are what you get in the `request.body`

**Comment by abhinav** [comment]
:+1:

File: GUIDE.rst [hunk 2]
```diff
@@ -144,10 +145,10 @@ so:
 .. code-block:: python
 
     def run():
-        app = TChannel('keyvalue-server')
-        app.register(KeyValue, handler=Get)
-        app.register(KeyValue, handler=Set)
-        app.listen()
+        tchannel = TChannel('keyvalue-server')
+        tchannel.thrift.register(KeyValue, handler=Get)
+        tchannel.thrift.register(KeyValue, handler=Set)
```
Comment (breerly) [inline GUIDE.rst]: this might be broken

Comment (breerly) [inline GUIDE.rst]: we need tests for calling register programatically

File: requirements-test.txt [hunk 3]
```diff
@@ -34,7 +34,8 @@ mock==1.0.1
 doubles
 
 # for releases
-zest.releaser
+zest.releaser[recommended]
+twine
```
Comment (breerly) [inline requirements-test.txt]: did zest move to using twine or something?

Comment (breerly) [inline requirements-test.txt]: or you uploaded after huh :)

File: GUIDE.rst (hunk 2 above)
Comment (breerly) [inline GUIDE.rst]: perhaps just remove this section for now

**Comment by breerly** [comment]
:shipit:

**Pull request merged**